asf_public_discourse_home_decarbonisation/pipeline/sentiment/sentence_sentiment.py

```diff
@@ -0,0 +1,199 @@
+
+python asf_public_discourse_home_decarbonisation/pipeline/sentiment/sentence_sentiment.py --source SOURCE --filter_by_expression FILTER_BY_EXPRESSION --start_date START_DATE --end_date END_DATE --relevant_clusters RELEVANT_CLUSTERS --irrelevant_clusters IRRELEVANT_CLUSTERS
+where
+- SOURCE is the source of the data e.g. "mse" or "buildhub"
+- [optional] FILTER_BY_EXPRESSION is the expression to filter by e.g. "heat pump"
+- [optional] START_DATE is the analysis start date in the format YYYY-MM-DD. Default to None (i.e. all data)
+- [optional] END_DATE is the analysis end date in the format YYYY-MM-DD. Defaults to None (all data)
+- [optional] RELEVANT_CLUSTERS is the clusters to keep e.g. "1,2,10"
+- [optional] IRRELEVANT_CLUSTERS is the clusters to remove e.g. "1,2,10"
+- [optional] add --test to run in test mode
+
+Examples for MSE:
+python asf_public_discourse_home_decarbonisation/pipeline/sentiment/sentence_sentiment.py --source "mse" --filter_by_expression "heat pump" --irrelevant_clusters "0,6" --start_date "2016-01-01" --end_date "2024-05-23"
+
+Examples for Buildhub:
+python asf_public_discourse_home_decarbonisation/pipeline/sentiment/sentence_sentiment.py --source "buildhub" --filter_by_expression "heat pump" --relevant_clusters "1,2,3,5,6,9,10,11,12,16,27,29,31,32,56,64" --start_date "2016-01-01" --end_date "2024-05-23"
+"""
+
+# Package imports
+from transformers import AutoModelForSequenceClassification
+from transformers import AutoTokenizer, AutoConfig
+import numpy as np
+from scipy.special import softmax
+from tqdm import tqdm
+from argparse import ArgumentParser
+from typing import Union
+import pandas as pd
+
+# Local imports
+from asf_public_discourse_home_decarbonisation.getters.getter_utils import (
+    save_to_s3,
+    load_s3_data,
+)
+from asf_public_discourse_home_decarbonisation.utils.general_utils import list_chunks
+from asf_public_discourse_home_decarbonisation import S3_BUCKET
+
+
+class SentenceBasedSentiment(object):
+    """Find sentiment scores for sentences"""
+
+    def __init__(
+        self,
+        model_name="cardiffnlp/twitter-roberta-base-sentiment-latest",
+    ):
+        """
+        Args:
+            model_name (str, optional): Model name. Defaults to "cardiffnlp/twitter-roberta-base-sentiment-latest".
+        """
+        self.model_name = model_name
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            model_name, use_fast=True, model_max_length=512
+        )
+        self.config = AutoConfig.from_pretrained(model_name)
+        self.model = AutoModelForSequenceClassification.from_pretrained(model_name)
+
+    def get_sentence_sentiment(self, texts: Union[list, str]) -> list:
+        """
+        Get the sentiment for a list of sentences using a pretrained sentiment model.
+        Returns a list of tuples with sentence, sentiment and the score:
+        e.g. ['sentence AAA', 'neutral', 0.6559918), ('sentence BBB', 'negative', 0.6319174)]
```

Comment:
```suggestion
        e.g. [("This is a really great sentence", 'positive', 0.97741115), ("This sentence is awful", 'negative', 0.9255473), ("Cat", 'neutral', 0.6470574)]
```
Think the example you used above is more informative, I'd find this "sentence AAA" slightly confusing.

Reply:
good shout :)